types/vimeo/index.d.ts

```diff
@@ -59,7 +59,8 @@ export class Vimeo {
      *                                  `host`, `port`, `query` or `headers`.
      * @param callback  Called when complete, `function (err, json)`.
      */
-    request(url: string | RequestOptions, callback: CompleteCallback): void;
+    // tslint:disable-next-line no-unnecessary-generics
+    request<T>(url: string | RequestOptions, callback: CompleteCallback<T>): void;
```

Comment:
The fact that you have to disable the lint rule tells you that this is a bit of an antipattern. Using uninferrable generics is just a trickier way of writing an unsafe downcast. For the very TypeScript-savvy, it’s a more compact syntax, but for folks who are coming to TypeScript from languages with reflection like C# or Java, this pattern looks dangerously like it _does something_ when it doesn’t. I am pretty strongly against this change (and all no-unnecessary-generics exemptions), but won’t stop you outright as the sole author of these definitions—instead I’ll just hope I can convince you that this has the potential to do more harm than good.

Reply:
Yeah, that makes sense. I've removed the generic and just typed the callback response `any` (8e49bd4) and I'll assert the type at the call sites.